src/google/google-extractor/main.py

```diff
@@ -34,62 +36,85 @@ def get_credentials():
     )
 
 
-# TODO: handle 'PARTIAL_DATA_AVAILABLE' warning
-# TODO: handle pagination
+@retry(
+    retry_on_exceptions=(Exception),
+    max_calls_total=4,
+    retry_window_after_first_call_in_seconds=60,
+)
+def execute(executable_resource):
+    return executable_resource.execute()
+
+
+# resource_method is the get/list SDK function to call,
+# resource_parameters is the parameters for get/list
+# response_property is the property in the response json we want
+# results is the result accumulated across pages
+@tail_recursive
+def request_execution(
+    resource_method, resource_parameters, response_property, results=None
+):
```

Comment:
There are several good styles to adopt for class and function documentation. Perhaps [numpydoc](https://numpydoc.readthedocs.io/en/latest/) would be a good one to choose.

```suggestion
@tail_recursive
def request_execution(
    resource_method, resource_parameters, response_property, results=None
):
    """ Execute a REST API request
   
    Parameters:
    ----------
    resource_method: list
        is the get/list SDK function to call
    resource_parameters: list
        is the parameters for get/list
    response_property: string
        is the property in the response json we want
    results: list
        is the result accumulated across pages   
    """
```

Adjust the data types if I interpreted incorrectly :-).

Reply:
Thanks, this was definitely the next step here.